src/ParametricOptInterface.jl

```diff
@@ -835,25 +843,113 @@ function MOI.optimize!(model::ParametricOptimizer)
         end
         empty!(model.updated_parameters)
     end
+
     MOI.optimize!(model.optimizer)
+
+    if model.evaluate_duals
+        calculate_dual_of_parameters(model)
+    end
 end
 
 ### Duals
 
-function MOI.get(model::ParametricOptimizer, attr::MOI.ConstraintDual, ci::MOI.ConstraintIndex)
-    return MOI.get(model.optimizer, attr, ci)
+function create_param_dual_cum_sum(keys_model_parameters::Base.KeySet{MOI.VariableIndex, Dict{MOI.VariableIndex, T}}) where T
+    param_dual_cum_sum = Dict{Int, Float64}()
+
+    for vi in keys_model_parameters
+        param_dual_cum_sum[vi.value] = 0.0
+    end
+    
+    return param_dual_cum_sum
+end
+
+function calculate_dual_of_parameters(model::ParametricOptimizer)
+    
+    param_dual_cum_sum = create_param_dual_cum_sum(keys(model.parameters))
+    
+    for S in ScalarLinearSetList
+        affine_constraint_cache_inner = model.affine_constraint_cache[MOI.ScalarAffineFunction{Float64}, S{Float64}]
+
+        # TODO
+        # Verify if !is_empty(model.affine_objective_cache) to pursue ?
+        if length(affine_constraint_cache_inner) > 0
+            # param_dual_cum_sum = iterate_over_constraint_cache_affine(model, model.affine_constraint_cache[MOI.ScalarAffineFunction{Float64}, S{Float64}] ,param_dual_cum_sum)
+            param_dual_cum_sum = iterate_over_constraint_cache_affine(model, affine_constraint_cache_inner ,param_dual_cum_sum)
+        end
+
+        quadratic_constraint_cache_pc_inner = model.quadratic_constraint_cache_pc[MOI.ScalarQuadraticFunction{Float64}, S{Float64}]
+
+        if length(quadratic_constraint_cache_pc_inner) > 0
+            param_dual_cum_sum = iterate_over_constraint_cache_quadratic(model, quadratic_constraint_cache_pc_inner, param_dual_cum_sum)
+        end
+    end
```

Comment:
To me, the point here is that you can have function barriers and check whether the dictionary is empty inside each function. Also, you hard-coded this `ScalarLinearSet`.

In my opinion the best option you be to have something like this.
```
con_types = MOI.get(model, MOI.ListOfConstraints())
update_duals_with_affine_constraint_cache!(param_dual_cum_sum, model, con_types)
update_duals_with_quadratic_constraint_cache!(param_dual_cum_sum, model, con_types)
```
and then make the loop inside each function with the appropriate function barriers